test/parse-errors.sh

```diff
@@ -1200,6 +1200,115 @@ ysh_case() {
   }
   '
 
+  # Example valid cases from grammar brain-storming
+  _ysh-should-parse '
+  case (add(10, 32)) {
+    (40 + 2) { echo Found the answer }
+    else { echo Incorrect
+    }
+  }
+  '
+  _ysh-should-parse "
+  case (file) {
+    / dot* '.py' /
+    {
+      echo Python
+    }
+
+    / dot* ('.cc' | '.h') /
+    {
+      echo C++
+    }
+  }
+  "
+  _ysh-should-parse '
+  case (lang) {
+      en-US
+    | en-CA
+    | en-UK {
+      echo Hello
+    }
+    fr-FR |
+    fr-CA {
+      echo Bonjour
+    }
+    else {
```

Comment:
Although it's a bit unusual, I was thinking `(else)` because it's less likely to be confused with `'else'`

The bare `else` word is a bit ambiguous here I think, because say `if` or `elif` are bare words

`(else)` is not a valid expression too, so we can "take over" that syntax for something related


---

On Zulip I suggested the possibility of adding a grammar production

```
matcher = '(' 'else' | testlist ')'
```